### DISCUSSION TITLE
ADB restore Prefs.xml

### DISCUSSION THREAD

**Discussion by yozart** [AnySoftKeyboard/AnySoftKeyboard#2980] [Q&A]
Hello guys,

I'm installing several devices and would like to set up Anysoft Keyboard as the default keyboard.

Currently, I'm using following commands to install and set the keyboard as default but, **How can I restore the AnySoftKeyboardPrefs.xml without opening the app on each device ?**

So far : 
`/*Send config file*/`
`adb push   "2-APPLICATIONS/AnySoftKeyboard/Data/." /sdcard/Android/data/com.menny.android.anysoftkeyboard/files/ > NUL`
`/*Install App*/`
`adb install -r -g "2-APPLICATIONS/AnySoftKeyboard/AnySoftKeyboard.com.apk" > NUL`
`/*Install language pack*/`
`adb install -r -g "2-APPLICATIONS/AnySoftKeyboard/FrenchAnySoftKeyboard_v4.0.1396.com.apk" > NUL`

**Comment by yozart** [comment]
Anyone has an answer? Is it feasible or just not in the plans ?

**Reply by menny** [reply]
I've never thought about such use case, but I guess it is for preinstalls.
I would say that the right way to do that is to compile a new version of AnySoftKeyboard with the desired defaults. But, I don't know if that's an option for you

**Reply by yozart** [reply]
Thank you for your answer, this solution should be a litle outside my current abilities unfortunately.
Do you think it would be possible for you to import automatically a config file if provided in a specific location ?
